plugin-api/src/main/java/org/sonar/api/batch/sensor/issue/IssueLocation.java

```diff
@@ -48,4 +49,10 @@ public interface IssueLocation {
   @CheckForNull
   String message();
 
+  /**
+   * @return list of text ranges in the message that can be highlighted when
+   * displaying the message according to specified type
+   */
+  List<MessageHighlighting> messageHighlighting();
```

Comment:
I see code highlighting as a specific use case, would the naming "MessageFormatting" make more sense?

Reply:
+1

Reply:
done